presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/ResponseWrapper.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.teradata.prestomanager.controller;
+
+import com.fasterxml.jackson.annotation.JsonGetter;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public final class ResponseWrapper
+{
+    private int statusCode;
+    private Status status;
+    private String entity;
+    private MultivaluedMap<String, Object> headers;
+
+    private ResponseWrapper(Response response)
+    {
+        this.statusCode = requireNonNull(response.getStatus());
+        this.status = requireNonNull(response.getStatusInfo().toEnum());
+        this.entity = response.hasEntity() ? response.getEntity().toString() : null;
+        this.headers = response.getHeaders().isEmpty() ? null : response.getHeaders();
+    }
+
+    public static ResponseWrapper wrapResponse(Response response)
+    {
+        return new ResponseWrapper(response);
+    }
+
+    public static List<ResponseWrapper> wrapResponseList(List<Response> responseList)
+    {
+        List<ResponseWrapper> returnList = new ArrayList<>();
+        for (Response response : responseList) {
+            returnList.add(new ResponseWrapper(response));
+        }
+        return returnList;
+    }
+
+    @JsonGetter("status")
+    public int getStatusCode()
+    {
+        return statusCode;
+    }
+
+    @JsonGetter("statusInfo")
+    public Status getStatus()
+    {
+        return status;
+    }
+
+    @JsonGetter("entity")
+    public Object getEntity()
```

Comment:
Yes. But you store it as String. If you want to return as an Object then store entity as Object.